### ISSUE TITLE
Ubuntu 18.04; PHP configuration paths reference PHP5

### ISSUE THREAD

**Issue by MattWilcox** [webmin/webmin#1194]
Webmin > Others > PHP Configuration

Only one item is listed, and it is wrong. There should also be lists here for /etc/php/7.2/fpm/php.ini etc.

Clicking the setting cog top left shows paths that all reference PHP5 which is not part of Ubuntu 18.04

**Comment by chris001** [comment]
You're right, it does appear that Webmin fails to detect (and use) PHP7.x config file locations.
Also, It's possible to have more than one PHP version config files under `/etc/php`, and still have remnants of the old PHP5.x config files installed under `/etc/php5`.  
Look on Debian 8, where I've uninstalled the EOL obsolete php5, and installed current php 7.3 :
```
# ls /etc/php
5.6  7.0  7.1  7.2  7.3  7.4
# ls /etc/php5
apache2  cgi  cli  conf.d  fpm  mods-available
```
Here's the "Webmin > Others > PHP Configuration" page, it fails to detect PHP7.x configs:
![webmin fails to detect php7 config locations](https://user-images.githubusercontent.com/259416/73756296-526dfc80-4735-11ea-8744-1b7a036fb6da.png)

**Comment by jcameron** [comment]
Do you also have Virtualmin installed, and if so does it detect PHP 7.3 when you run a config re-check?

**Comment by MattWilcox** [comment]
I have VirtualMin installed also, and it detects PHP7.3 - but the bug where it also detects 7.3 twice remains; On an initial install, after running through the config wiard, check-configuration, it says something about 7.3 and 7.3 (duplicate version) and re-generating php.ini's ... it makes no difference to the above report. PHP is pretty broken in Ubuntu tbh, especially if you're wanting to use FPM - which ought to be the default.

**Comment by rostovtsev** [comment]
> something about 7.3 and 7.3 (duplicate version)

That has been fixed already and will be included in the next version of Virtualmin.

Jamie, 

  1. After I installed multiple versions of PHP on Ubuntu 18, like:

~~~
5.6  7.0  7.1  7.4
~~~

.. and re-run config re-check on Virtualmin side the new config files _did not_ seem to appear automatically in PHP configuration module. Is this meant to be? 

  2. I don't remember, if I mentioned it already, and if it was fixed but server-owners see global PHP configurations files with **Manage** and **Edit Manually** buttons/links in the table:
     ![image](https://user-images.githubusercontent.com/4426533/73832422-2e83e700-4819-11ea-965f-d1f9395c5f3e.png)
    When clicked, and trying to save the file it gives expected - permission denied error. I think, you meant to allow server owners to check for global configuration, if needed, which is great but maybe **Manage** button should go and **Edit Manually** should just turn to **View Configuration** button?

**Comment by jcameron** [comment]
The fix that was included in Webmin 1.980 will only be picked up in new installs. 

If you've upgraded, you can fix the config to pick up all PHP versions by replacing `/etc/webmin/phpini/config` with 👍 

```
php_ini=/etc/php*/apache2/php.ini,/etc/php/*/apache2/php.ini=Configuration for mod_php	/etc/php*/cgi/php.ini,/etc/php/*/cgi/php.ini=Configuration for scripts run via CGI	/etc/php*/cli/php.ini,/etc/php/*/cli/php.ini=Configuration for command-line scripts
```

**Comment by iliarostovtsev** [comment]
> The fix that was included in Webmin 1.980 will only be picked up in new installs.

Jamie, can we create a facility to cherry-pick Webmin modules configs updates in post-install?

**Comment by arnoldnijboer** [comment]
Thanks, that worked.
would be nice if the script would be updated as webmin gets updated.
